ingestors/pypi_xml_rpc.go

```diff
@@ -29,13 +51,67 @@ func (ingestor *PyPiXmlRpc) Schedule() string {
 	return "@every 5m"
 }
 
+// Structured storage for the tuple returned by the xmlrpc client
+type PyPiXmlRpcResponse struct {
+	Name      string
+	Version   string
+	Timestamp int64
+	Action    string
+}
+
+// Return trhe if this response is an ingestable action
+func (response *PyPiXmlRpcResponse) IsIngestionAction() bool {
+	switch response.Action {
+	case "new release", "yank release", "remove release":
+		return true
+	}
+
+	return false
+}
+
+// Get the PackageVersion struct for this response
+func (response *PyPiXmlRpcResponse) GetPackageVersion() data.PackageVersion {
+	createdAt := time.Unix(response.Timestamp, 0)
+	discoveryLag := time.Since(createdAt)
+
+	return data.PackageVersion{
+		Platform:     "pypi",
+		Name:         response.Name,
+		Version:      response.Version,
+		CreatedAt:    createdAt,
+		DiscoveryLag: discoveryLag,
+	}
+}
+
+// Validate and then create a PyPiXmlRpcResponse from a log struct
+func createResponseStruct(log []interface{}) (*PyPiXmlRpcResponse, error) {
```

Comment:
golang has the `any` keyword as an alias to `interface{}`, JIC you want it to more semantically explicit